v-next/hardhat/src/internal/cli/init/package-manager.ts

```diff
@@ -0,0 +1,153 @@
+import { execSync } from "node:child_process";
+import path from "node:path";
+
+import { exists } from "@ignored/hardhat-vnext-utils/fs";
+import semver from "semver";
+
+type PackageManager = "npm" | "yarn" | "pnpm";
+
+/**
+ * getPackageManager returns the name of the package manager used in the workspace.
+ * It determines this by checking the presence of package manager specific lock files.
+ *
+ * @param workspace The path to the workspace where the package manager should be checked.
+ * @returns The name of the package manager used in the workspace.
+ */
+export async function getPackageManager(
+  workspace: string,
+): Promise<PackageManager> {
+  const pathToYarnLock = path.join(workspace, "yarn.lock");
+  const pathToPnpmLock = path.join(workspace, "pnpm-lock.yaml");
+
+  if (await exists(pathToYarnLock)) {
+    return "yarn";
+  }
+  if (await exists(pathToPnpmLock)) {
+    return "pnpm";
+  }
+  return "npm";
+}
+
+/**
+ * getDevDependenciesInstallationCommand returns the command to install the given dependencies
+ * as dev dependencies using the given package manager.
+ *
+ * @param packageManager The package manager to use.
+ * @param dependencies The dependencies to install.
+ * @returns The installation command.
+ */
+export function getDevDependenciesInstallationCommand(
+  packageManager: PackageManager,
+  dependencies: string[],
+): string[] {
+  const packageManagerToCommand: Record<PackageManager, string[]> = {
+    npm: ["npm", "install", "--save-dev"],
+    yarn: ["yarn", "add", "--dev"],
+    pnpm: ["pnpm", "add", "--save-dev"],
+  };
+  const command = packageManagerToCommand[packageManager];
+  command.push(...dependencies);
+  return command;
+}
+
+/**
+ * installsPeerDependenciesByDefault returns true if the package manager
+ * installs peer dependencies by default.
+ *
+ * @param workspace The path to the workspace where the package manager will operate.
+ * @param packageManager The package manager to use.
+ * @param version The version of the package manager to use. This parameter is used only for testing.
+ * @param config The configuration of the package manager to use. This parameter is used only for testing.
+ * @returns True if the package manager installs peer dependencies by default, false otherwise.
+ */
+export async function installsPeerDependenciesByDefault(
+  workspace: string,
+  packageManager: PackageManager,
+  version?: string,
+  config?: Record<string, string>,
+): Promise<boolean> {
+  switch (packageManager) {
+    case "npm":
```

Comment:
Oh, nice, I thought this function was going to be `packageManager !== 'yarn'`. But this is more robust.